### ISSUE TITLE
noProcessEnv should report env imported from node:process

### ISSUE THREAD

**Issue by dBianchii** [biomejs/biome#10619]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.0

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  NODE_PACKAGE_MANAGER:         bun/1.3.13
```

</details>

### Rule name

`style/noProcessEnv`

### Reproduction

Use this config:

```json
{
  "formatter": { "enabled": false },
  "assist": { "enabled": false },
  "linter": {
    "rules": {
      "recommended": false,
      "style": {
        "noProcessEnv": "error"
      }
    }
  }
}
```

Given this file:

```ts
import { env } from "node:process";

const fromNodeProcessImport = env["TEST_ENV"];

void fromNodeProcessImport;
```

Run:

```bash
biome check node-process-env.ts
```

### Current result

No diagnostic is emitted on `@biomejs/biome@2.5.0`.

### Expected result

`style/noProcessEnv` should report this as a direct `process.env` read through the named `env` export from `node:process`.

### Workaround

Consumers can add a `noRestrictedImports` entry for named `env` imports from `node:process` and `process`, but this seems semantically owned by `noProcessEnv`.

### Related

This is a narrower follow-up to #10447. That issue is closed, but this specific `node:process` named import case still reproduces on Biome 2.5.0. The separate `Bun.env` discussion is tracked in #10617 because maintainers noted that `Bun.env` may belong in a different rule.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
